### DISCUSSION TITLE
RE2 Release Map Requirements

### DISCUSSION THREAD

**Discussion by no-lex** [redeclipse/base#1185] [General]
Release candidates for RE2's beta launch need to be submitted November 1st to be fully verified before release in December.

What does your map need to be a release candidate?

### General guidelines
The map needs to generally follow the guidelines laid out in arena shooter layout guides, including having three exits per room and limited clear line of sight. Mappers should take extra care to make maps with less than 100m straight corridors and avoid long clear lines of sight. 

A typical duel map takes up approximately an entire gridsize 10 map and a typical medium CTF map takes up approximately an entire gridsize 11 map.

### Technical requirements
1. Performance requirements:
 i. At `mapeffects 1` must be at least 50 fps at all locations on Kaveri A8/ HD 630 graphics at 1080p, minimum `smsize` and `smfilter` with `rhtaps 20`.
 ii. Each mapeffects level should be more intensive than the one below it. Recommend at least 30% performance drop  between `mapeffects 1` and `mapeffects 3`. 
 iii. Analysis of occlusion and fixing of simple occlusion issues where possible
2. Layout and Map Construction
 i. Pickups finalized (this requires many hours of playtest and modification)
 ii. Pickups not construed to favor any weapon in particular
 iii. Pickup areas appropriately paletted to indicate which weapon is where
 iv. Clipping complete and navigating the map is maximally hangup-free (also requires iteration in playtest), including sky texture clipping if needed
 v. Bot waypoints complete and verified to not have pathing errors
 vi. Scenery geometry present and complete 
 vii. Rooms unambiguous and not prone to being lost in (visual clues should tell player where they are at all times)
 viii. Map balanced to accomodate all weapon combinations and not to especially favor particular types of combat universally
3. Texturing is complete and detailed
 i. Windows have consistent border texturing
 ii. Insets and extrusions consistently and properly textured on their sides
 iii. Tiles not cut off half way
 iv. Back faces of folded cubes skytextured and out-of-scenery faces flattened or skytextured
 v. Tiling of organic textures subverted in some way (if applicable)
4. Geometry is detailed and high poly 
 i. Highest possible curve resolution present whenever in playable area
 ii. Minimum possible geometry in areas not visible
 iii. Geometry not placed in the way of typical player paths
5. Lighting appropriately set for all mapeffects levels
 i. All out of scene lighting should be off at minimum `mapeffects`
 ii. Lighting should not favor any mapeffects level (not too dark at minimum settings e.g.)
 iii. Maximum lighting should not disable shadows on lights that can plausibly cast them; the 
whole point of the engine is dynamic lighting
 iv. Lighting must cover all of map; no areas darker than 0x202020 under any circumstance
 v. Specular highlights should be physically possible (please do not have downward facing spotlights casting ceiling speculars)
6. Mapmodels
 i. Out of scene mapmodels disabled at `mapeffects 1`
 ii. In-scene mapmodels verified for clipping issues
 iii. Mapmodels appropriate to texturing and theme of the rest of the map (do not just add it because it's in the list)
7. Appropriate support for both normal and alt modes
 i. Lighting appropriate to both modes, including lights only enabled on one mode if necessary
 ii. Performance for the two variants should be fairly close to each other (within 30%)
 iii. Alt mode should be immediately discernible from normal mode
8. Sounds complete and present
 i. Map music defined for level
 ii. Portals and jumpers required to have discernible sounds that can be heard from a large part of the map
 iii. Ambiance sounds appropriate to the scene, with care not to overuse obnoxious or repetitive sounds over large areas; may require different sounds for different map variants.
9. Affinities and Team Maps
 i. Team maps provided with at least an affinity in each base and one in the center of the map
 ii. No more than 5 DAC points for maps with playable areas smaller than gridpower 11
 iii. Player spawns provided for each base
 iv. Balanced (heavily recommended to be symmetrical) weapon placement
 v. Obvious map indications on which side of map player is on at all times
10. General map requirements
 i. Map author field set
 ii. License explicitly stated, clear, and valid for inclusion (CC-BY-SA or more liberal)
 iii. Map desciption provided
 iv. Unique map title provided, no longer than 16 characters
 v. Map file name reasonably close to map title
 vi. Several player spawns provided for each player the map expects to accomodate

Most of these are not rocket science and most experienced mappers already hold themselves to these points without giving it a second thought. However, explicitly setting standards before hand is important to prevent surprises from arising at the deadline.

**Comment by Calinou** [comment]
Not sure if I should create a dedicated issue for this, but I noticed item spawners are currently clipped:

![2019](https://user-images.githubusercontent.com/180032/71325606-85469300-24ef-11ea-9e4a-e5b6f8c30ca0.jpg)

Due to how stair smoothing works in Red Eclipse, this is kind of distracting and slows down the player a bit. Shouldn't they be noclipped instead? (I'm currently looking at clipping issues in the official maps and will open a PR soon.)

Also, in the particular case above on `institute`, this causes the player to become "stuck" in some cases due to the low ceiling.
